commands/build.go

```diff
@@ -692,6 +692,11 @@ func wrapBuildError(err error, bake bool) error {
 			msg += " Named contexts are supported since Dockerfile v1.4. Use #syntax directive in Dockerfile or update to latest BuildKit."
 			return &wrapped{err, msg}
 		}
+		if st.Code() == codes.Unimplemented && strings.Contains(st.Message(), "unsupported frontend capability moby.buildkit.frontend.gitquerystring") {
+			msg := "current frontend does not support Git URLs with query string components."
+			msg += " Git URLs with query string are supported since Dockerfile v1.18 and BuildKit v0.24. Use BUILDKIT_SYNTAX build-arg, #syntax directive in Dockerfile or update to latest BuildKit."
```

Comment:
Does this require _both_ the updated frontend _and_ latest BuildKit, or is updating the frontend sufficient? The error is a bit ambiguous (using an older frontend with BuildKit v0.24 would still fail I guess? But not sure if just updating the syntax would fix that).

Recommending users to update to latest BuildKit also won't work if this error is returned from the default builder in dockerd; perhaps we should have a `/go/` link to the Dockerfile syntax directive (which could link to alternatives such as the `BUILDKIT_SYNTAX`.

Reply:
> perhaps we should have a `/go/` link to the Dockerfile syntax directive (which could link to alternatives such as the `BUILDKIT_SYNTAX`.

Do you mean a `/go` link in the error message with more information? Was thinking of putting some of these use cases https://github.com/crazy-max/buildx-buildkit-tests/tree/main/buildkit-6178#readme along https://github.com/docker/docs/pull/23322